modules/security-context-constraints-about.adoc

```diff
@@ -8,6 +8,11 @@
 
 Similar to the way that RBAC resources control user access, administrators can use _security context constraints_ (SCCs) to control permissions for pods. These permissions include actions that a pod can perform and what resources it can access. You can use SCCs to define a set of conditions that a pod must run with to be accepted into the system.
 
+[IMPORTANT]
+====
+There is a difference in permissions experienced by containers, when you create pods directly by `kubectl` or `oc`. The pod creation uses the `kubectl` or `oc` user, but the deployment falls back to the `ServiceAccount`.
```

Comment:
All of the mentions of `kubectl` and `oc` should be removed, the client is irrelevant. I can create pods directly with `curl` just as well.

```suggestion
There is a difference in permissions experienced by containers when you create pods directly, compared to these being created by pod controllers such as `Deployment`. Direct pod creation may use the user's permissions to determine the applicable SCC but the deployment always uses permissions of its `ServiceAccount`.
```

I think you should not apply the above suggestion by @cardil, I am not aware of deployments, jobs, replicasets being called "higher level types". Also, the part about "exceeding safe OpenShift standards" is quite misleading.

Reply:
@stlaz @cardil Shall we rewrite like this ? 
`When a ServiceAccount is not specified in the pod definition, OpenShift Container Platform uses the default ServiceAccount to deploy the pod. The default ServiceAccount uses the restricted SCC. Setting the pod to run with a UIDs outside the range of the namespace requires assigning the anyuid SCC to the ServiceAccount used to deploy the Pod.`

Source: https://cloud.redhat.com/blog/a-guide-to-openshift-and-uids

Reply:
I think this isn't true. That's the whole point of this issue.

@darshan-nagaraj wrote:

> _When a ServiceAccount is not specified in the pod definition, OpenShift Container Platform uses the default ServiceAccount to deploy the pod._

No. OCP will use user context, likely kubeadmin, instead of service account.

> _The default ServiceAccount uses the restricted SCC._

That used context, likely kubeadmin, can use unrestricted SCC. The pod will run with UID 0.

For every other resource type, (deployment, job, replica set) the above would be true, but not if user creates a pod directly.

Reply:
@darshan-nagaraj @cardil  As the admission control section already mentions
```The set of SCCs that admission uses to authorize a pod are determined by the user identity and groups that the user belongs to. Additionally, if the pod specifies a service account, the set of allowable SCCs includes any constraints accessible to the service account.```

Can we duplicate this text to the current PR?, this does make a lot more sense to me

Reply:
@y4sht Sure, I can add it.
@cardil @stlaz Do you agree with the changes ?

Reply:
> I think this isn't true. That's the whole point of this issue.
You're wrong. The point of this issue is that direct pod creation has permissions of UNION(user, serviceAccount), whereas pods deployment by a controller only have permissions of the service account.

> No. OCP will use user context, likely kubeadmin, instead of service account.
here, too, the quote is in fact correct

> That used context, likely kubeadmin, can use unrestricted SCC. The pod will run with UID 0.
the pod uses the permissions of UNION(user, serviceAccount)

@darshan-nagaraj Perhaps if it is unclear, see if you can get the above detail in. @y4sht's description is not 100% accurate either. The term "user identity" already includes the group information so ` and groups that the user belongs to` is redundant. On the other hand, it does not mention the service account name. Plus, it says that you have to specify a SA for a pod but by default, the `default` SA for a given NS is applied, as you mentioned.